### PR TITLE
Add log_cosh loss.

### LINKED ISSUE #103: Implement log(cosh()) loss function
log-cosh is a doubly differentiable alternative to the huber loss. A naive implementation is prone to overflow (since cosh has an e^x term), so I think it'd be a useful addition to the library. Plus, it's implemented in other libraries, such as [TensorFlow](https://www.tensorflow.org/api_docs/python/tf/keras/losses/log_cosh).

If this sounds like a relevant addition, I'd be happy to contribute it!

### PR DESCRIPTION
**Pull request by n2cholas**
Closes #103.

Notes:
- The link to "How to Train Neural Networks" by Ralph Neuneier and Hans Georg Zimmermann is not a free resource. I am not aware of any open access references for log-cosh except for https://openreview.net/forum?id=rkglvsC9Ym (which I'm not sure about using since it's an openreview link and not the original source).
- I didn't include the `a > 0` hyperparameter since other implementations don't (namely TensorFlow and TensorFlow Probability).

### CODE REVIEW

**Comment by mtthss** [comment]
Thanks for the contribution!
I think the open review link is better since its open access

**Comment by n2cholas** [comment]
Done, updated the reference to `[Chen et al, 2019](https://openreview.net/pdf?id=rkglvsC9Ym)`!

**Review by mtthss — APPROVED** [review]

**Comment by mtthss** [comment]
Thanks!

**Review by mtthss — APPROVED** [review]

**Review by mtthss — APPROVED** [review]
